src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.HandleArray.cs

```diff
@@ -27,23 +27,16 @@ private static void HandleStartArray(JsonSerializerOptions options, ref ReadStac
             {
                 jsonPropertyInfo = state.Current.JsonClassInfo.CreateRootProperty(options);
             }
-            else if (state.Current.JsonClassInfo.ClassType == ClassType.Unknown)
```

Comment:
This is unreachable (dead code) because this case is handled in: https://github.com/dotnet/corefx/blob/4c3bd8a0533a2e35df371b91045a1a5f0ada7d54/src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.cs#L105-L109